### DISCUSSION TITLE
Cant find path to datasets

### DISCUSSION THREAD

**Discussion by baukenieuwenhuis** [tudelft/AE4353-Y24#16] [Ex_1]
In the notebook ex 1, the yaml file connects to the path:

 /workspaces/AE4353-Y24/data/2D_QUAD_HOVER.npz

however the path which I copied into the .json file is:

/home/baukenieuwenhuis/ae4353_dataset

the error I get is in the third cell when trying to prepare the data:
<img width="689" alt="error ai" src="https://github.com/user-attachments/assets/ab09e736-0bc7-4b4a-bb93-499a9b9a6fab">

Should I change the path in the .json file? or is there an underlying problem.

**Comment by yilun-wu** [comment]
Do you have `2D_QUAD_HOVER.npz` under `/home/baukenieuwenhuis/ae4353_dataset`?

**Reply by baukenieuwenhuis** [reply]
Yes that's the pwd path I copied following the readme, but it still says it cant find it

**Comment by Huizerd** [comment]
can you paste your `devcontainer.json` and `2d.yaml` here?

**Comment by baukenieuwenhuis** [comment]
Yaml:

trajectory_dim: 2
dataset_file: /workspaces/AE4353-Y24/data/2D_QUAD_HOVER.npz

batchsize_train: 1024
batchsize_val: 4096
shuffle_train: True
shuffle_val: False
num_workers_train_loader: 0
num_workers_val_loader: 0

model_hidden_dims: 120
model_num_layers: 4

epochs: 2
validate_per_steps: 500
lr: 1.0 # to be tuned

json file:
{
    "name": "ae4353",
    "build": {
        "dockerfile": "./Dockerfile",
        "context": ".."
    },
    "mounts": [
        // Bind mount data here if necessary e.g.
        //source=/home/baukenieuwenhuis/ae4353_dataset,target=${containerWorkspaceFolder}/data,type=bind
    ],
    "features": {
        "ghcr.io/devcontainers/features/common-utils:2": {
            "installZsh": "true",
            "username": "vscode",
            "userUid": "1000",
            "userGid": "1000",
            "upgradePackages": "true"
        },
        "ghcr.io/devcontainers/features/git:1": {
            "version": "latest",
            "ppa": "false"
        }
    },
    "customizations": {
        // Configure properties specific to VS Code.
        "vscode": {
            "settings": {
                // TODO: change to specific conda env python bin
                "python.defaultInterpreterPath": "/opt/conda/envs/AE4353/bin/python",
                "python.linting.enabled": true,
                "jupyter.widgetScriptSources": [
                    "jsdelivr.com",
                    "unpkg.com"
                ],
                "notebook.defaultFormatter": "ms-python.black-formatter",
                "notebook.formatOnSave.enabled": true,
                "python.terminal.activateEnvironment": true,
                "[python]": {
                    "editor.defaultFormatter": "ms-python.black-formatter",
                    "editor.formatOnSave": true
                }
            },
            "extensions": [
                "ms-python.python",
                "ms-python.vscode-pylance",
                "ms-python.black-formatter",
                "ms-toolsai.jupyter",
                "ms-toolsai.tensorboard",
                "github.copilot",
                "github.copilot-chat"
            ]
        }
    },
    "hostRequirements": {
        "gpu": "optional"
    },
    // Uncomment the below "runArgs" section if you intend to use a GPU.
    // "runArgs": [
    //     "--gpus",
    //     "all"
    // ],
    "postCreateCommand": "./.devcontainer/postCreateCommand.sh",
    // Use 'forwardPorts' to make a list of ports inside the container available locally.
    // "forwardPorts": [],
    // Use 'postCreateCommand' to run commands after the container is created.
    // "postCreateCommand": "uname -a",
    // Set `remoteUser` to `root` to connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
    "remoteUser": "vscode",
    "remoteEnv": {
        "SERVICE_HOST": "host.docker.internal",
        "SERVICE_PORT": "9876"
    }
}

**Reply by Huizerd** [reply]
this line
```
//source=/home/baukenieuwenhuis/ae4353_dataset,target=${containerWorkspaceFolder}/data,type=bind
```

is still commented out (see the `//` at the start. it should look like
```
source=/home/baukenieuwenhuis/ae4353_dataset,target=${containerWorkspaceFolder}/data,type=bind
```
and then it should work!

**Reply by Huizerd** [reply]
you can also see this from the fact that it has a dark grey color in the vscode window, which indicates that it's a comment (like the one above saying `// Bind mount data...`

**Reply by baukenieuwenhuis** [reply]
//"source=<your-external-data-directory>,target=${containerWorkspaceFolder}/data,type=bind"

this is the original line, should I remove "" as well or should they stay? that's what i confused with commented line

**Reply by Huizerd** [reply]
it should look like this (with your folder name obviously)
![image](https://github.com/user-attachments/assets/77dbe855-ac29-4fdb-bcea-5c2997d6975c)

so the quotes are necessary. they indicate that the whole thing is a string (like in python), which we want to send to the container. the `//` are like python's `#` for a comment

**Comment by baukenieuwenhuis** [comment]
<img width="715" alt="devcontainer" src="https://github.com/user-attachments/assets/be475e8b-1a8b-4383-9578-7a075efb7ad8">
Tried to rebuild the json file after fixing the line but now I get this error

**Reply by baukenieuwenhuis** [reply]
The first time the container built without any issues^

**Reply by Huizerd** [reply]
someone else had these issues too. these packages are for security but not necessary for the course. so if you make your `Dockerfile` look like this:
![image](https://github.com/user-attachments/assets/9d2bec23-95ed-4da2-9c2e-d4978438aa76)
commenting out lines 9-13 so it won't try to install these packages

**Reply by baukenieuwenhuis** [reply]
Still get the following message:
<img width="706" alt="devcontainer" src="https://github.com/user-attachments/assets/2c5281d6-0d85-473d-9da7-00a9b0a8e68e">

**Reply by Huizerd** [reply]
can you rebuild the container without cache?

**Reply by baukenieuwenhuis** [reply]
Gives the exact same error message

**Reply by Huizerd** [reply]
we can have a look today!
